Lib/dbm/sqlite3.py

```diff
@@ -0,0 +1,145 @@
+import os
+import sqlite3
+import sys
+from pathlib import Path
+from contextlib import suppress, closing
+from collections.abc import MutableMapping
+
+BUILD_TABLE = """
+  CREATE TABLE IF NOT EXISTS Dict (
+    key TEXT UNIQUE NOT NULL,
+    value BLOB NOT NULL
+  )
+"""
+GET_SIZE = "SELECT COUNT (key) FROM Dict"
+LOOKUP_KEY = "SELECT value FROM Dict WHERE key = ?"
+STORE_KV = "REPLACE INTO Dict (key, value) VALUES (?, ?)"
+DELETE_KEY = "DELETE FROM Dict WHERE key = ?"
+ITER_KEYS = "SELECT key FROM Dict"
+
+
+class error(OSError):
+    pass
+
+
+_ERR_CLOSED = "DBM object has already been closed"
+_ERR_REINIT = "DBM object does not support reinitialization"
+
+
+def _normalize_uri(path):
+    path = os.fsdecode(path)
+    path = Path(path)
+    uri = path.absolute().as_uri()
+    while "//" in uri:
+        uri = uri.replace("//", "/")
+    return uri
+
+
+class _Database(MutableMapping):
+
+    def __init__(self, path, /, flag):
+        if hasattr(self, "_cx"):
+            raise error(_ERR_REINIT)
+
+        match flag:
+            case "r":
+                flag = "ro"
+            case "w":
+                flag = "rw"
+            case "c":
+                flag = "rwc"
+            case "n":
+                flag = "rwc"
+                try:
+                    os.remove(path)
+                except FileNotFoundError:
+                    pass
+            case _:
+                raise ValueError("Flag must be one of 'r', 'w', 'c', or 'n', "
+                                 f"not {flag!r}")
+
+        # We use the URI format when opening the database.
+        uri = _normalize_uri(path)
+        uri = f"{uri}?mode={flag}"
+
+        try:
+            self._cx = sqlite3.connect(uri, autocommit=True, uri=True)
+        except sqlite3.Error as exc:
+            raise error(str(exc))
+
+        # This is an optimization only; it's ok if it fails.
+        with suppress(sqlite3.OperationalError):
+            self._cx.execute("PRAGMA journal_mode = wal")
+
+        if flag == "rwc":
+            with closing(self._execute(BUILD_TABLE)):
+                pass
+
+    def _execute(self, *args, **kwargs):
+        if not self._cx:
+            raise error(_ERR_CLOSED)
+        try:
+            ret = self._cx.execute(*args, **kwargs)
```

Comment:
It can be simply:
```suggestion
            return self._cx.execute(*args, **kwargs)
```
It is even better to wrap it in `closing()`:
```suggestion
            return closing(self._cx.execute(*args, **kwargs))
```

Reply:
Wrapping in closing is a nice trick; I will apply that.